iot_verifier/src/poc.rs

```diff
@@ -658,6 +675,44 @@ fn verify_gw_capability(is_full_hotspot: bool) -> GenericVerifyResult {
     Ok(())
 }
 
+/// verify witness lag
+/// if the first received event is the beacon then,
+/// all witnesses must be received within MAX_BEACON_TO_WITNESS_LAG milliseconds of the beacon
+/// if the first received event is a witness then,
+/// all subsequent witnesses must be received within MAX_WITNESS_LAG milliseconds of that first witness
+fn verify_witness_lag(
+    beacon_received_ts: DateTime<Utc>,
+    first_witness_ts: DateTime<Utc>,
+    received_ts: DateTime<Utc>,
+) -> GenericVerifyResult {
+    let (first_event_ts, max_permitted_lag) = if beacon_received_ts <= first_witness_ts {
+        (beacon_received_ts, MAX_BEACON_TO_WITNESS_LAG)
+    } else {
+        (first_witness_ts, MAX_WITNESS_LAG)
+    };
+    let this_witness_lag = received_ts
```

Comment:
Why not just subtract the datetimes and then compare Durations rather than converting everything to a timestamp?

Reply:
that goes back to an earlier draft where the witness could be 4s before or after the beacon.  For that i used abs diff which returns a u64 and thus defaulted all comparison params to u64s to accommodate.

I dont actually need the abs diff anymore, will update

Reply:
addressed here: fdf9cd4a3b756cc9da400d524e41f26d20e615a9